aea/aea_builder.py

```diff
@@ -1811,12 +1812,16 @@ def _find_import_order(
                 dependencies, component_type = configuration.skills, SKILLS
             elif isinstance(configuration, ConnectionConfig):
                 dependencies, component_type = configuration.connections, CONNECTIONS
+            elif isinstance(configuration, ContractConfig):
+                dependencies, component_type = configuration.contracts, CONTRACTS
             else:
                 raise AEAException("Not a valid configuration type.")  # pragma: nocover
-            for dependency in dependencies:
-                dependency_to_supported_dependencies[
-                    ComponentId(ComponentType.SKILL, dependency)
-                ].add(component_id)
+
+            if component_type != CONTRACTS:
+                for dependency in dependencies:
+                    dependency_to_supported_dependencies[
+                        ComponentId(ComponentType.SKILL, dependency)
+                    ].add(component_id)
```

Comment:
I don't understand this block. Why is the type hardcoded for skill and both used when evaluating a skill or connection? Why is contract excluded?

Reply:
I'm still trying to understand the purpose of this block. I excluded this block because it caused an error while building to make the tests pass temporarily. This block maps every dependency from a given component to the skill type and It caused the build to break. To test this hypothesis I created some dummy connections and made them interdependent and the build failed with the same error with which the build for contracts failed. 

```
Error: Package (skill, default_author/c0:0.1.0) not found.
```
c0 is a connection here.

Reply:
I'm still trying to make some sense of this block, The last commit was from you (https://github.com/valory-xyz/open-aea/commit/64f729b2a2e2770af23fc197549208da617fc0a7#r62348671) (see if you can help with it)

Reply:
```
            for dependency in dependencies:
                dependency_to_supported_dependencies[
                    ComponentId(component_type[:-1], dependency)
                ].add(component_id)
 ```

Reply:
I think this is correct, so without the if condition you added.